Packages/com.unity.inputsystem/Tests/InputSystem/Plugins/SwitchTests.cs

```diff
@@ -49,16 +71,133 @@ public void Devices_SupportsSwitchNpad()
+                    return InputDeviceCommand.kGenericFailure;
+                });
+        }
+        Assert.That(controller.npadId, Is.EqualTo(NPad.NpadId.Handheld));
+        Assert.That(controller.orientation, Is.EqualTo(NPad.Orientation.Vertical));
+        Assert.That(controller.styleMask, Is.EqualTo(NPad.NpadStyle.Handheld));
+    }
+
+    [Test]
+    [Category("Devices")]
+    public unsafe void Devices_CanSetControllerOrientation()
+    {
+        var controller = InputSystem.AddDevice<NPad>();
+
+        NpadDeviceIOCTLSetOrientation? receivedCommand = null;
+        unsafe
+        {
+            testRuntime.SetDeviceCommandCallback(controller.id,
+                (id, commandPtr) =>
+                {
+                    if (commandPtr->type == NpadDeviceIOCTLSetOrientation.Type)
+                    {
+                        Assert.That(receivedCommand.HasValue, Is.False);
+                        receivedCommand = *((NpadDeviceIOCTLSetOrientation*)commandPtr);
+                        return 1;
+                    }
+
+                    Assert.Fail("Received wrong type of command");
+                    return InputDeviceCommand.kGenericFailure;
+                });
+        }
+        controller.SetOrientationToSingleJoyCon(NPad.Orientation.Horizontal);
+
+        Assert.That(receivedCommand.HasValue, Is.True);
+        Assert.That(receivedCommand.Value.orientation, Is.EqualTo(NPad.Orientation.Horizontal));
+
+        receivedCommand = null;
+        controller.SetOrientationToSingleJoyCon(NPad.Orientation.Vertical);
+
+        Assert.That(receivedCommand.HasValue, Is.True);
+        Assert.That(receivedCommand.Value.orientation, Is.EqualTo(NPad.Orientation.Vertical));
+    }
+
+    [Test]
+    [Category("Devices")]
+    public unsafe void Devices_CanStartSixAxisSensors()
+    {
+        var controller = InputSystem.AddDevice<NPad>();
+
+        NpadDeviceIOCTLStartSixAxisSensor? receivedCommand = null;
+        unsafe
+        {
+            testRuntime.SetDeviceCommandCallback(controller.id,
+                (id, commandPtr) =>
+                {
+                    if (commandPtr->type == NpadDeviceIOCTLStartSixAxisSensor.Type)
+                    {
+                        Assert.That(receivedCommand.HasValue, Is.False);
+                        receivedCommand = *((NpadDeviceIOCTLStartSixAxisSensor*)commandPtr);
+                        return 1;
+                    }
+
+                    Assert.Fail("Received wrong type of command");
+                    return InputDeviceCommand.kGenericFailure;
+                });
+        }
+        controller.StartSixAxisSensor();
+
+        Assert.That(receivedCommand.HasValue, Is.True);
+    }
+
+    [Test]
+    [Category("Devices")]
+    public unsafe void Devices_CanStopSixAxisSensors()
+    {
+        var controller = InputSystem.AddDevice<NPad>();
+
+        NpadDeviceIOCTLStopSixAxisSensor? receivedCommand = null;
+        unsafe
```

Comment:
Minor but this is redundant. If the method is unsafe, all code in the body can use unsafe features.